client/src/accounts/authentication/register.hook.test.tsx

```diff
@@ -0,0 +1,117 @@
+import { describe, expect, it } from 'vitest';
+import { HTTP_BAD_REQUEST, HTTP_OK } from '~/api/api.constants';
+import { rest, server } from '~/mocks/server';
+import { act, renderHook, waitFor } from '~/test/utils';
+import { RegistrationFormType, useRegister } from './register.hook';
+
+const ENDPOINT = '*/api/accounts/register/';
+
+interface Result {
+  id: Number;
+  error?: { message: string };
+  mutate: (form: Record<string, unknown>) => void;
+  isError: boolean;
+  isSuccess: boolean;
+  data?: Record<string, unknown>;
+}
+
+describe('useRegisterHook', () => {
+  it('should throw an error if any fields are missing', async () => {
+    // we're missing the confirm password
+    const expectedMessage = 'Please re-authenticate';
+    server.use(
+      rest.post(ENDPOINT, (req, res, ctx) => res(ctx.status(HTTP_BAD_REQUEST), ctx.json({ message: expectedMessage }))),
+    );
+
+    const registerForm: RegistrationFormType = {
+      email: 'bob@example.com',
+      firstName: 'Bob',
+      lastName: 'Ross',
+      password: 'otherpassword',
+      confirmPassword: 'otherpassword',
+    };
+    const { result } = renderHook<Result, RegistrationFormType>(() => useRegister());
+
+    act(() => {
+      result.current.mutate(registerForm);
+    });
+    await waitFor(() => {
+      expect(result.current.isError).toBe(true);
+      expect(result.current.error).toEqual({ message: expectedMessage });
+    });
```

Comment:
This doesn't follow the pattern I pinned (see last pinned comment) to the dev channel, is there a reason for this? I expected to see

```
await waitFor(() => expect(result.current.isSuccessful).toBe(true);

expect(result.current.isError).toBe(true);
expect(result.current.error).toEqual({ message: expectedMessage });
```

It is a common mistake, according to Kent Dodds to have multiple assertions in a `waitFor`, read this [blog](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback). I've linked to the exact section, but please re-read the whole blog, might fill in some gaps in your knowledge.